3rdParty/iresearch/core/store/data_input.hpp

```diff
@@ -146,7 +146,14 @@ struct IRESEARCH_API index_input : public data_input {
   virtual int64_t checksum(size_t offset) const = 0;
 
  private:
+  #if defined(__clang__) && __clang_major__ >= 13
+   #pragma GCC diagnostic push
```

Comment:
I guess that for \_\_clang\_\_ we should use "clang" instead of "GCC". See example from 3rdParty/iresearch/core/iql/parser_context.cpp:819

```
parser_context::query_node const& parser_context::find_node(
  parser::semantic_type const& value
) const {
  #if defined(__APPLE__)
    #pragma clang diagnostic push
    #pragma clang diagnostic ignored "-Wtautological-compare"
  #elif defined (__GNUC__)
    #pragma GCC diagnostic push
    #pragma GCC diagnostic ignored "-Wtype-limits"
  #elif defined(_MSC_VER)
    #pragma warning(disable: 4127) // conditional expression is constant
  #endif

    // parser::semantic_type may be defined as a signed value in parser.yy
    if (std::is_signed<parser::semantic_type>::value && value < 0) {
      return m_nodes[0];
    }

  #if defined(__APPLE__)
    #pragma clang diagnostic pop
  #elif defined (__GNUC__)
    #pragma GCC diagnostic pop
  #elif defined(_MSC_VER)
    #pragma warning(default: 4127)
  #endif

  return value < m_nodes.size() ? m_nodes[value] : m_nodes[0];
}
```

Reply:
Fixed.